app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt

```diff
@@ -204,146 +167,58 @@ open class DefaultToolbarMenu(
     } ?: false
     // End of predicates //
 
-    val installToHomescreen = BrowserMenuHighlightableItem(
-        label = context.getString(R.string.browser_menu_install_on_homescreen),
-        startImageResource = R.drawable.ic_add_to_homescreen,
-        iconTintColorResource = primaryTextColor(),
-        highlight = BrowserMenuHighlight.LowPriority(
-            label = context.getString(R.string.browser_menu_install_on_homescreen),
-            notificationTint = getColor(context, R.color.whats_new_notification_color)
-        ),
-        isHighlighted = {
-            !context.settings().installPwaOpened
-        }
+    private val installPwaToHomescreen = toolbarMenuItems.installPwaToHomescreen
+    private val newTabItem = toolbarMenuItems.newTabItem
+    private val historyItem = toolbarMenuItems.historyItem
+    private val downloadsItem = toolbarMenuItems.downloadsItem
+    private var findInPageItem = toolbarMenuItems.findInPageItem
+    private var desktopSiteItem = toolbarMenuItems.requestDesktopSiteItem
+    private var customizeReaderView = toolbarMenuItems.customizeReaderView
+    private var openInApp = toolbarMenuItems.openInAppItem
+    private var addToHomeScreenItem = toolbarMenuItems.addToHomeScreenItem
+    private var addToTopSitesItem = toolbarMenuItems.addToTopSitesItem
+    private var saveToCollectionItem = toolbarMenuItems.saveToCollectionItem
+    private var settingsItem = toolbarMenuItems.settingsItem
+    private var deleteDataOnQuit = toolbarMenuItems.deleteDataOnQuitItem
+    private var syncedTabsItem = toolbarMenuItems.oldSyncedTabsItem
+    private var syncSignInItem = toolbarMenuItems.syncSignInItem
+
+    private val extensionsItem = WebExtensionPlaceholderMenuItem(
+        id = WebExtensionPlaceholderMenuItem.MAIN_EXTENSIONS_MENU_ID
+    )
+
+    private val reportSiteIssuePlaceholder = WebExtensionPlaceholderMenuItem(
+        id = WebCompatReporterFeature.WEBCOMPAT_REPORTER_EXTENSION_ID
+    )
+
+    private var addEditBookmarksItem = BrowserMenuImageTextCheckboxButton(
+        imageResource = R.drawable.ic_bookmarks_menu,
+        iconTintColorResource = primaryTextColor,
+        label = context.getString(R.string.library_bookmarks),
+        labelListener = {
+            onItemTapped.invoke(Item.Bookmarks)
+        },
+        primaryStateIconResource = R.drawable.ic_bookmark_outline,
+        secondaryStateIconResource = R.drawable.ic_bookmark_filled,
+        tintColorResource = accentBrightTextColor,
+        primaryLabel = context.getString(R.string.browser_menu_add),
+        secondaryLabel = context.getString(R.string.browser_menu_edit),
+        isInPrimaryState = { !isCurrentUrlBookmarked }
     ) {
-        onItemTapped.invoke(ToolbarMenu.Item.InstallPwaToHomeScreen)
+        handleBookmarkItemTapped()
     }
 
     private val oldCoreMenuItems by lazy {
```

Comment:
Do we still need to maintain this old list of menu items since the pref is flipped? I think we could clean up some of the code if we are already shipping if not perhaps we should wait until we can delete the preference and all of the old menu items before landing the refactor. We typically forget to remove the secret preference and prefs after a feature is pref on by default for all releases.